cmd/controller/controller.go

```diff
@@ -430,8 +431,18 @@ func (c *command) start(ctx context.Context) error {
 		c.ClusterComponents.Add(ctx, metrics)
 	}
 
-	if !slices.Contains(c.DisableComponents, constant.KubeletConfigComponentName) {
-		c.ClusterComponents.Add(ctx, controller.NewKubeletConfig(c.K0sVars, adminClientFactory))
+	if !slices.Contains(c.DisableComponents, constant.WorkerConfigComponentName) {
```

Comment:
Let's keep ident level closer to the main code flow and split the branches:
- one if branch for instantiating and adding workerconfig reconciler
- one if branch for logging warning

Something like:

```go
if !slices.Contains(c.DisableComponents, constant.WorkerConfigComponentName) {
	reconciler, err := workerconfig.NewReconciler(c.K0sVars, c.NodeConfig.Spec, adminClientFactory, leaderElector)
	if err != nil {
		return err
	}
	c.ClusterComponents.Add(ctx, reconciler)
}
if slices.Contains(c.DisableComponents, constant.KubeletConfigComponentName) {
		logrus.Warnf("Usage of deprecated component name %q, please switch to %q",
				constant.KubeletConfigComponentName, constant.WorkerConfigComponentName,
			)
}
```

Reply:
I think in this case, the deprecated label wouldn't disable the component, would it?

Reply:
if it is a necessity, (though personal I think it is better to fail with validation message during the arg parse step) we could modify c.DisableComponents accordingly by appending proper component name there.

Reply:
Moving this to a validation step makes absolute sense. I think accepting/rewriting the old flag and issuing a warning is the best way for a deprecation. If k0s would stop accepting the old value, this would be a breaking change. I also discovered that some other components were missing from the list of available components. I opened #2474 for those things.

Reply:
Done by normalizing the deprecated component name to the new component name in the config package.